pkg/api/rest/create.go

```diff
@@ -57,7 +57,14 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx api.Context, obj runtime.Obje
 			return errors.NewBadRequest("the namespace of the provided object does not match the namespace sent on the request")
 		}
 	} else {
-		objectMeta.Namespace = api.NamespaceNone
+		if ns, ok := api.NamespaceFrom(ctx); ok && ns != api.NamespaceNone {
+			return errors.NewBadRequest("namespace is not allowed on this type")
```

Comment:
Please note that some types (events) can have both namespaced and non-namespaced objects. Does this handle that case?